File: tests/benchmark/list/list.js
function reverse(list) {
	var i=0, tmp, idx, count=list.length;
	for (; i < count; i++) {
		idx = count - i - 1;
		tmp = list[idx];
		list[idx] = list[i];
		list[i] = tmp;
	}
}

var start = process.hrtime();

var i=0, N=20000000, list=Array(N);
for (; i < N; i++) list[i] = i;
reverse(list);

var end = process.hrtime(start);
var secs = (end[0] + end[1] / 1e9).toFixed(6) + 's';
console.log('elapsed:', secs);

File: tests/benchmarks/list/list.lua
local function reverse(arr)
	local len = #arr
	local max = math.floor(len / 2)

	for i=1, max do
		local idx = len + 1 - i
		arr[i], arr[idx] = arr[idx], arr[i]
	end
end

local N = 20000000
local list = {}
for i=1, N do
	list[i] = i
end

local start = os.clock()
reverse(list)
local seconds = os.clock() - start
print('elapsed: ' .. seconds .. 's')
